packages/loaders/src/Dots.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { retrieveXTransform, retrieveYTransform, KEYFRAME_MAX } from './utils/dot-transforms';
+import { LoadingPlaceholder, StyledCircle, StyledSVG } from './styled-elements';
+
+export default class Dots extends React.Component {
+  static propTypes = {
+    /** Size of the loader. Can inherit from `font-size` styling. */
+    size: PropTypes.any,
+    /** Velocity (speed) of the animation. Between -1 and 1. */
+    velocity: PropTypes.number,
+    /** Color of the loader. Can inherit from `color` styling. */
+    color: PropTypes.string,
+    /**
+     * Delay in MS to begin loader rendering. This helps prevent
+     * quick flashes of the loader during normal loading times.
+     **/
+    delayMS: PropTypes.number
+  };
+
+  static defaultProps = {
+    size: 'inherit',
+    color: 'inherit',
+    velocity: 0,
+    delayMS: 750
+  };
+
+  state = {
+    frame: 0,
+    delayComplete: false,
+    timestamp: 0
+  };
+
+  componentDidMount() {
+    const { delayMS } = this.props;
+
+    this.renderingDelayTimeout = setTimeout(() => {
+      this.setState({ delayComplete: true }, () => {
+        this.performAnimationFrame();
+      });
+    }, delayMS);
+  }
+
+  componentWillUnmount() {
+    clearTimeout(this.renderingDelayTimeout);
+    cancelAnimationFrame(this.animationFrame);
+  }
+
+  performAnimationFrame = (timestamp = 0) => {
+    const { velocity } = this.props;
```

Comment:
Does it make sense to constraint this value to be in the -1:1 range here?

Reply:
This could require a custom prop-validator. Is it better to pin these internally in this method?

Reply:
Yeah, I was thinking that if the value was out-of-range, we just pin it to -1 (minimum) or 1 (maximum).

Reply:
These are now pinned within their bounds.